unified_planning/plans/plan.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2021 AIPlan4EU project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+import unified_planning as up
+import unified_planning.model
+from unified_planning.model import FNode, Action, InstantaneousAction, Expression, Effect
+from unified_planning.walkers import Substituter, Simplifier
+from typing import Callable, Dict, Optional, Set, Tuple, List
+from fractions import Fraction
+
+
+'''This module defines the different plan classes.'''
+
+
+class ActionInstance:
+    '''Represents an action instance with the actual parameters.'''
+    def __init__(self, action: 'unified_planning.model.Action', params: Tuple['unified_planning.model.FNode', ...] = tuple()):
+        assert len(action.parameters) == len(params)
+        self._action = action
+        self._params = tuple(params)
+
+    def __repr__(self) -> str:
+        s = []
+        if len(self._params) > 0:
+            s.append('(')
+            first = True
+            for p in self._params:
+                if not first:
+                    s.append(', ')
+                s.append(str(p))
+                first = False
+            s.append(')')
+        return self._action.name + ''.join(s)
+
+    @property
+    def action(self) -> 'Action':
+        '''Returns the action.'''
+        return self._action
+
+    @property
+    def actual_parameters(self) -> Tuple['FNode', ...]:
+        '''Returns the actual parameters.'''
+        return self._params
+
+def ground_action_instance(action_instance: ActionInstance, substituter: 'Substituter', simplifier: 'Simplifier') -> 'Action':
```

Comment:
document!

Reply:
Actually, I was thinking about moving the logic of an Action grounding into the Action class itself. This might be also used by the Grounder, what do you think about this?